design/kanister-kopia-integration.md

```diff
@@ -0,0 +1,471 @@
+kind: RepositoryServer
+metadata:
+   name: repository-monitoring
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+spec:
+   location:
+      # required: storage location info
+      secret:
+         name: location
+         namespace: kanister
+      # required: creds to access the location
+      credentialSecret:
+         name: loc-creds
+         namespace: kanister
+   repository:
+      # repository must be created manually before creating server CR
+      # required: path for the repository - will be relative sub path
+      # within the path prefix specified in the location
+      rootPath: /repo/monitoring/
+      # required: password to access the repository
+      passwordSecret:
+         name: repository-monitoring-password
+         namespace: kanister
+      # required: these values will be used by the repository server controller
+      # for the connection from the server to the repository. Kopia will use
+      # these values to track and control operations
+      username: kanisterAdmin
+      hostname: monitoring.app
+   server:
+      # required: server admin details
+      adminSecret:
+         name: repository-monitoring-server-admin
+         namespace: kanister
+      # required: TLS certificate required for secure communication between the Kopia client and server
+      tlsSecret:
+         name: repository-monitoring-server-tls
+         namespace: kanister
+      # required: repository users list
+      userAccessSecret:
+         name: repository-monitoring-server-access
+         namespace: kanister
+      # required: selector for network policy required to enable cross-namespace access to the server
+      labelSelector:
+         matchLabels:
+         - key1: value1
+         - key2: value2
+status:
+   conditions:
+   - lastTransitionTime: "2022-08-20T09:48:36Z"
+     lastUpdateTime: "2022-08-20T09:48:36Z"
+     status: "True"
+     type: RepositoryServerReady
+   serverInfo:
+      podName: "repository-monitoring-pod-a1b2c3"
+      networkPolicyName: "repository-monitoring-np-d4e5f6"
+      serviceName: "repository-monitoring-svc-g7h8i9"
+```
+
+The required `spec.location.secret` refers to a `Secret` resource storing location-related sensitive data. This secret is provided by the user.
+For example,
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+   name: location
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: Opaque
+data:
+   # required: supported values are s3, gcs, and azure
+   type: s3
+   # required
+   bucket: my-bucket
+   # optional: specified in case of S3-compatible stores
+   endpoint: https://foo.example.com
+   # optional: used as a sub path in the bucket for all backups
+   path: kanister/backups
+   # required, if supported by the provider
+   region: us-west-1
+   # optional: if set to true, do not verify SSL cert.
+   # Default, when omitted, is false
+   skipSSLVerify: false
```

Comment:
We should document all the supported keys. to help the user to put this secret together. AFAICT, the only place where this is documented now is in the `kopia` CLI [code](https://github.com/kopia/kopia/blob/05e729a7858a6e86cb48ba29fb53cb6045efce2b/cli/storage_s3.go#L19-L27).